src/main.rs

```diff
@@ -64,37 +44,37 @@ fn compress(source_directory: String, temp_directory: String) -> String {
     };
     let enc = GzEncoder::new(tar_gz, Compression::default());
     let mut tar = tar::Builder::new(enc);
-    match tar.append_dir_all("", &source_directory) {
+    match tar.append_dir_all("", &config.source) {
         Ok(()) => (),
         Err(error) => {
-            error!("Unable to append dir {}.  Returned error is {}", &source_directory, error);
+            error!("Unable to append dir {}.  Returned error is {}", &config.source, error);
             tar.finish().expect("Unable to close temp file");
-            delete_file(filename);
+            delete_file(&filename);
             panic!("Unable to locate dir.");
         }
     }
     tar.finish().unwrap();
 
     println!("Compression success");
 
-    return filename;
+    filename
```

Comment:
Pas de `return` ?

Reply:
Par défaut tout block (fonction, condition, boucle ou simple `{}`) à pour valeur sa dernière expression

https://rustwiki.org/en/book/ch03-03-how-functions-work.html#statements-and-expressions